integrations/aws/aws/auth_provider.py

```diff
@@ -0,0 +1,187 @@
+        self, application_session: aioboto3.Session
+    ) -> List[Dict[str, Any]]:
+        """
+        Returns a list of all accessible AWS accounts as dictionaries with 'Id' and 'Name'.
+        """
+        pass
+
+    @abstractmethod
+    async def get_account_credentials(
+        self, sts_client: STSClient, account: Dict[str, Any]
+    ) -> Optional[AwsCredentials]:
+        """
+        Assume a role in the given account and return the credentials.
+        """
+        pass
+
+    def expiry_time(self) -> str:
+        expiry = datetime.now(timezone.utc) + timedelta(
+            seconds=ASSUME_ROLE_DURATION_SECONDS
+        )
+        return expiry.isoformat()
+
+
+class ApplicationCredentialsProvider(CredentialsProvider):
+    async def get_application_credentials(self) -> AwsCredentials:
+        aws_access_key_id = ocean.integration_config.get("aws_access_key_id")
+        aws_secret_access_key = ocean.integration_config.get("aws_secret_access_key")
+        if not aws_access_key_id or not aws_secret_access_key:
+            logger.warning(
+                "No AWS access key ID or secret access key provided, trying to fetch credentials using default boto chain."
+            )
+
+        credentials = {
+            "aws_access_key_id": aws_access_key_id,
+            "aws_secret_access_key": aws_secret_access_key,
+        }
+
+        default_session = aioboto3.Session(**credentials)
+        async with default_session.client("sts") as sts_client:
+            caller_identity = await sts_client.get_caller_identity()
+            current_account_id = caller_identity["Account"]
+            default_credentials = await default_session.get_credentials()  # type: ignore
+            frozen_credentials = await default_credentials.get_frozen_credentials()
+            return AwsCredentials(
+                account_id=current_account_id,
+                sts_client=sts_client,
+                access_key_id=frozen_credentials.access_key,
+                secret_access_key=frozen_credentials.secret_key,
+                session_token=frozen_credentials.token,
+            )
+
+    async def get_all_accessible_accounts(
+        self, application_session: aioboto3.Session
+    ) -> List[Dict[str, Any]]:
+        # For just the application creds provider, we only return the current account
+        async with application_session.client("sts") as sts_client:
+            caller_identity = await sts_client.get_caller_identity()
+            return [{"Id": caller_identity["Account"], "Name": "No name found"}]
+
+    async def get_account_credentials(
+        self, sts_client: STSClient, account: Dict[str, Any]
+    ) -> Optional[AwsCredentials]:
+        # For application-only scenario, there's no cross-account role assumption.
+        # Just return the application credentials again if needed.
+        raise NotImplementedError(
+            "No cross-account assumption for application-only scenario."
+        )
```

Comment:
I don't think this method should be abstracted if only OrganizationCredentialsProvider can use it